### PR TITLE
Fixes #12535 - expose config_groups in ENC

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by ares** [comment]
Could you please add a test that also config groups of host's host group and its parent host group are included? Thank you.

**Comment by shlomizadok** [comment]
Added. Thanks :see_no_evil:

**Comment by dLobatog** [comment]
Ready to merge if Jenkins tests pass (@ares the tests now include a test for the hostgroup config group too)

**Comment by ares** [comment]
I need to test it first. I'll merge afterwards

**Comment by ares** [comment]
Ok, works fine, just last thing. We tend to prefix our custom ENC values by "foreman" prefix, since they can collide with fact and other parameters. I think "foreman_config_groups" name would be better. Other than that, :+1:

**Comment by shlomizadok** [comment]
@ares, reworded the variable name. Thanks

File: test/unit/host_test.rb [hunk 1]
```diff
@@ -2359,6 +2359,23 @@ def to_managed!
     assert enc['parameters']['foreman_subnets'].any? {|s| s['network_type'] == 'IPv6'}
   end
 
+  test "#info ENC YAML contains config_groups" do
+    host = FactoryGirl.build(:host)
+    host.config_groups = [config_groups(:one)]
+    enc = host.info
+    assert_includes(enc.keys, 'config_groups')
```
Comment (ares) [inline test/unit/host_test.rb]: oops, s/config_groups/foreman_config_groups/ :-)

**Comment by shlomizadok** [comment]
@ares, shouldn't this be APJ?

**Comment by ares** [comment]
> @ares, shouldn't this be APJ?

nope, sorry I failed to finish the review (first time I used new github feature), good to merge after fixing tests

File: test/unit/host_test.rb (hunk 1 above)
Comment (shlomizadok) [reply test/unit/host_test.rb]: I'm sorry, missed that. Fixed.

**Comment by ares** [comment]
APJ

**Comment by dLobatog** [comment]
[test foreman]

**Comment by shlomizadok** [comment]
[test foreman] pretty please?

**Comment by dLobatog** [comment]
Thanks @ares and @shlomizadok !

**Pull request merged**